File: TwentyFourDuel-Android/app/src/main/java/com/twentyfourduel/ui/PlayerView.kt
package com.twentyfourduel.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.SpanStyle
import androidx.compose.ui.text.buildAnnotatedString
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.twentyfourduel.model.GamePhase
import com.twentyfourduel.model.GameState
import com.twentyfourduel.model.Token

@Composable
fun PlayerView(game: GameState, playerIndex: Int, modifier: Modifier = Modifier) {
    val player = game.getPlayer(playerIndex)
    val accent = if (playerIndex == 0) AppColors.p1Accent else AppColors.p2Accent
    val name = if (playerIndex == 0) "玩家一" else "玩家二"
    val isRoundOver = game.phase == GamePhase.ROUND_OVER
    val otherWantsEnd = game.getPlayer(1 - playerIndex).wantsToEnd

    Box(modifier = modifier.fillMaxWidth().background(
        Brush.radialGradient(listOf(accent.copy(alpha = 0.025f), Color.Transparent), radius = 400f)
    )) {
        Column(
            modifier = Modifier.fillMaxSize().padding(horizontal = 16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(Modifier.height(4.dp))

            // Info bar
            Row(
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier.fillMaxWidth()
            ) {
                Box(Modifier.size(10.dp).clip(CircleShape).background(accent))
                Spacer(Modifier.width(6.dp))
                Text(name, fontSize = 15.sp, fontWeight = FontWeight.Bold, color = Color.White)
                Spacer(Modifier.width(6.dp))
                EndGameButton(player.wantsToEnd) { game.handleRequestEnd(playerIndex) }
                if (otherWantsEnd && !player.wantsToEnd) {
                    Spacer(Modifier.width(4.dp))
                    Text("对方请求结束", fontSize = 11.sp, color = AppColors.gold)
                }
                Spacer(Modifier.weight(1f))
                if (game.timeLimit == 0) {
                    SkipButton(game, playerIndex)
                    Spacer(Modifier.width(4.dp))
                }
                Text("${player.score}", fontSize = 20.sp, fontWeight = FontWeight.ExtraBold, color = accent)
            }

            // Round / Timer bar
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(
                    "第 ${game.currentRound} 轮 / 共 ${game.totalRounds} 轮",
                    fontSize = 13.sp, fontWeight = FontWeight.SemiBold, color = AppColors.textDim
                )
                Spacer(Modifier.width(12.dp))
                Text("⏱", fontSize = 11.sp, color = AppColors.textDim)
                Spacer(Modifier.width(4.dp))
                if (game.timeLimit > 0) {
                    val tc = if (game.timeLeft <= 10) AppColors.errorRed else AppColors.gold
                    Text("${game.timeLeft}", fontSize = 18.sp, fontWeight = FontWeight.ExtraBold, color = tc, fontFamily = FontFamily.Monospace)
                } else {
                    Text("∞", fontSize = 18.sp, fontWeight = FontWeight.ExtraBold, color = AppColors.gold)
                }
            }
            Spacer(Modifier.height(4.dp))

            val interactAlpha = if (isRoundOver) 0.4f else 1f

            // Expression display
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .heightIn(min = 46.dp)
                    .clip(RoundedCornerShape(12.dp))
                    .background(AppColors.bgInput)
                    .border(2.dp, when {
                        player.showSuccess -> AppColors.successGreen
                        player.feedback != null && !player.passed -> AppColors.errorRed
                        else -> AppColors.borderDim
                    }, RoundedCornerShape(12.dp))
                    .padding(horizontal = 14.dp, vertical = 10.dp)
                    .alpha(interactAlpha),
                contentAlignment = Alignment.CenterStart
            ) {
                if (player.tokens.isEmpty()) {
                    Text("点击数字和运算符组成算式", fontSize = 14.sp, color = Color(0xFF2B2B2B))
                } else {
                    Text(
                        buildAnnotatedString {
                            player.tokens.forEach { token ->
                                val (color, weight) = when (token) {
                                    is Token.Number -> AppColors.tokenNum to FontWeight.Bold
                                    is Token.Op -> AppColors.tokenOp to FontWeight.Normal
                                    is Token.Paren -> AppColors.tokenParen to FontWeight.Normal
                                }
                                withStyle(SpanStyle(color = color, fontWeight = weight, fontFamily = FontFamily.Monospace)) {
                                    append(token.display)
                                }
                            }
                        },
                        fontSize = 26.sp,
                        maxLines = 1
                    )
                }
            }

            // Feedback
            Text(
                player.feedback ?: "",
                fontSize = 13.sp,
                color = AppColors.errorRed,
                modifier = Modifier.height(20.dp).alpha(if (player.feedback != null) 1f else 0f)
            )

            Spacer(Modifier.height(4.dp))

            // Number buttons
            Row(
                horizontalArrangement = Arrangement.spacedBy(10.dp),
                modifier = Modifier.alpha(interactAlpha)
            ) {
                for (i in 0 until 4) {
                    val used = player.usedCards.getOrElse(i) { false }
                    val canInput = game.canInputNumber(playerIndex)
                    val disabled = used || !canInput
                    val suit = game.suits.getOrElse(i) { com.twentyfourduel.model.CardSuit.SPADE }
                    val num = game.numbers.getOrElse(i) { 0 }
                    Box(
                        modifier = Modifier
                            .width(70.dp).height(88.dp)
                            .clip(RoundedCornerShape(12.dp))
                            .background(Brush.linearGradient(listOf(Color(28, 34, 64), Color(20, 26, 46))))
                            .border(2.dp, accent.copy(alpha = 0.15f), RoundedCornerShape(12.dp))
                            .alpha(if (used) 0.18f else if (disabled) 0.35f else 1f)
                            .clickable(enabled = !disabled && !isRoundOver) {
                                game.handleNumber(playerIndex, i)
                            },
                        contentAlignment = Alignment.Center
                    ) {
                        Column(horizontalAlignment = Alignment.CenterHorizontally) {
                            Text(suit.symbol, fontSize = 10.sp, color = Color.White.copy(alpha = 0.45f))
                            Text("$num", fontSize = 28.sp, fontWeight = FontWeight.Bold, color = Color.White)
                        }
                    }
                }
            }

            Spacer(Modifier.height(6.dp))

            // Operator buttons
            Row(
                horizontalArrangement = Arrangement.spacedBy(6.dp),
                modifier = Modifier.alpha(interactAlpha)
            ) {
                val canOp = game.canInputOp(playerIndex)
                val canOpen = game.canInputOpenParen(playerIndex)
                val canClose = game.canInputCloseParen(playerIndex)
                listOf("+" to "+", "-" to "−", "*" to "×", "/" to "÷", "(" to "(", ")" to ")").forEach { (op, label) ->
                    val enabled = when (op) {
                        "(" -> canOpen; ")" -> canClose; else -> canOp
                    }
                    Box(
                        modifier = Modifier
                            .width(50.dp).height(44.dp)
                            .clip(RoundedCornerShape(10.dp))
                            .background(AppColors.bgCard)
                            .border(2.dp, AppColors.borderDim, RoundedCornerShape(10.dp))
                            .alpha(if (enabled) 1f else 0.3f)
                            .clickable(enabled = enabled && !isRoundOver) {
                                game.handleOp(playerIndex, op)
                            },
                        contentAlignment = Alignment.Center
                    ) {
                        Text(label, fontSize = 21.sp, color = Color(152, 152, 216))
                    }
                }
            }

            Spacer(Modifier.height(6.dp))

            // Action buttons
            Row(
                horizontalArrangement = Arrangement.spacedBy(7.dp),
                modifier = Modifier.alpha(interactAlpha)
            ) {
                ActionBtn("提交", AppColors.successGreen, 88.dp, isRoundOver) { game.handleSubmit(playerIndex) }
                ActionBtn("清空", Color(120, 120, 120), 70.dp, isRoundOver) { game.handleClear(playerIndex) }
                ActionBtn("⌫", Color(120, 120, 120), 70.dp, isRoundOver) { game.handleBackspace(playerIndex) }
            }

            Spacer(Modifier.height(4.dp))
        }

        // Round over overlay
        if (isRoundOver) {
            Box(
                modifier = Modifier.fillMaxSize().background(AppColors.bgMain.copy(alpha = 0.7f)),
                contentAlignment = Alignment.Center
            ) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    val winner = game.roundWinner
                    if (winner != null) {
                        if (winner == playerIndex) {
                            Text("正确！+1 分", fontSize = 28.sp, fontWeight = FontWeight.ExtraBold, color = AppColors.successGreen)
                        } else {
                            Text("对手先答出", fontSize = 28.sp, fontWeight = FontWeight.ExtraBold, color = AppColors.errorRed)
                            game.solutions.firstOrNull()?.let {
                                Text(it, fontSize = 15.sp, color = AppColors.textDim)
                            }
                        }
                    } else {
                        Text("本轮结束", fontSize = 28.sp, fontWeight = FontWeight.ExtraBold, color = AppColors.gold)
                        game.solutions.firstOrNull()?.let {
                            Text("答案: $it", fontSize = 15.sp, color = AppColors.textDim)
                        }
                    }
                }
            }
        }
    }
}

@Composable
private fun EndGameButton(wantsToEnd: Boolean, onClick: () -> Unit) {
    val bg = if (wantsToEnd) AppColors.gold.copy(alpha = 0.12f) else AppColors.bgCard
    val border = if (wantsToEnd) AppColors.gold.copy(alpha = 0.4f) else AppColors.borderDim
    val tc = if (wantsToEnd) AppColors.gold else Color(128, 128, 128)
    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(8.dp))
            .background(bg)
            .border(1.dp, border, RoundedCornerShape(8.dp))
            .clickable(onClick = onClick)
            .padding(horizontal = 10.dp, vertical = 5.dp)
    ) {
        Text(if (wantsToEnd) "取消结束" else "结束", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = tc)
    }
}

@Composable
private fun SkipButton(game: GameState, playerIndex: Int) {
    val player = game.getPlayer(playerIndex)
    val otherSkip = game.getPlayer(1 - playerIndex).wantsToSkip
    val bg = if (player.wantsToSkip) AppColors.gold.copy(alpha = 0.12f) else AppColors.bgCard
    val border = if (player.wantsToSkip) AppColors.gold.copy(alpha = 0.4f) else AppColors.borderDim
    val tc = if (player.wantsToSkip) AppColors.gold else Color(128, 128, 128)
    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(8.dp))
            .background(bg)
            .border(1.dp, border, RoundedCornerShape(8.dp))
            .clickable { game.handleRequestSkip(playerIndex) }
            .padding(horizontal = 10.dp, vertical = 5.dp)
    ) {
        Row {
            Text(if (player.wantsToSkip) "取消跳过" else "跳过", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = tc)
            if (otherSkip && !player.wantsToSkip) {
                Spacer(Modifier.width(3.dp))
                Text("(对方已请求)", fontSize = 10.sp, color = tc)
            }
        }
    }
}

@Composable
private fun ActionBtn(
    title: String,
    color: Color,
    minW: androidx.compose.ui.unit.Dp,
    disabled: Boolean,
    onClick: () -> Unit
) {
    val borderColor = if (title == "提交") color.copy(alpha = 0.35f) else AppColors.borderDim
    Box(
        modifier = Modifier
            .widthIn(min = minW).height(44.dp)
            .clip(RoundedCornerShape(10.dp))
            .background(AppColors.bgCard)
            .border(2.dp, borderColor, RoundedCornerShape(10.dp))
            .clickable(enabled = !disabled, onClick = onClick),
        contentAlignment = Alignment.Center
    ) {
        Text(title, fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = color)
    }
}

File: TwentyFourDuel.swiftpm/Sources/PlayerView.swift
import SwiftUI

struct PlayerView: View {
    @ObservedObject var game: GameState
    let playerIndex: Int

    private var player: PlayerState { game.players[playerIndex] }
    private var accent: Color { playerIndex == 0 ? .p1Accent : .p2Accent }
    private var name: String { playerIndex == 0 ? "玩家一" : "玩家二" }
    private var isRoundOver: Bool { game.phase == .roundOver }

    var body: some View {
        ZStack {
            playerBackground

            VStack(spacing: 0) {
                Spacer(minLength: 4)
                infoBar
                roundTimerBar
                Spacer(minLength: 4)

                Group {
                    expressionDisplay
                    feedbackText
                    Spacer(minLength: 8)
                    numberButtons
                    Spacer(minLength: 6)
                    operatorButtons
                    Spacer(minLength: 6)
                    actionButtons
                }
                .allowsHitTesting(!isRoundOver)
                .opacity(isRoundOver ? 0.4 : 1)

                Spacer(minLength: 4)
            }
            .padding(.horizontal, 16)

            if isRoundOver { roundOverlay }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .clipped()
    }

    // MARK: - Info Bar

    private var otherWantsToEnd: Bool {
        game.players[1 - playerIndex].wantsToEnd
    }

    private var infoBar: some View {
        HStack(spacing: 6) {
            Circle()
                .fill(accent)
                .frame(width: 10, height: 10)
                .shadow(color: accent, radius: 4)
            Text(name)
                .font(.system(size: 15, weight: .bold))

            endGameButton

            if otherWantsToEnd && !player.wantsToEnd {
                Text("对方请求结束")
                    .font(.system(size: 11))
                    .foregroundColor(.gold)
            }

            Spacer()

            if game.timeLimit == 0 {
                skipButtonInBar
            }

            Text("\(player.score)")
                .font(.system(size: 20, weight: .heavy))
                .foregroundColor(accent)
                .contentTransition(.numericText())
                .animation(.spring(response: 0.3), value: player.score)
        }
        .frame(maxWidth: 560)
        .padding(.horizontal, 4)
    }

    private var endGameButton: some View {
        Button {
            game.handleRequestEnd(player: playerIndex)
        } label: {
            Text(player.wantsToEnd ? "取消结束" : "结束")
                .font(.system(size: 12, weight: .semibold))
                .foregroundColor(player.wantsToEnd ? .gold : Color(white: 0.5))
                .padding(.horizontal, 10)
                .padding(.vertical, 5)
                .background(
                    RoundedRectangle(cornerRadius: 8)
                        .fill(player.wantsToEnd ? Color.gold.opacity(0.12) : Color.bgCard)
                )
                .overlay(
                    RoundedRectangle(cornerRadius: 8)
                        .stroke(player.wantsToEnd ? Color.gold.opacity(0.4) : Color.borderDim, lineWidth: 1)
                )
        }
    }

    private var skipButtonInBar: some View {
        Button {
            game.handleRequestSkip(player: playerIndex)
        } label: {
            HStack(spacing: 3) {
                Text(player.wantsToSkip ? "取消跳过" : "跳过")
                    .font(.system(size: 12, weight: .semibold))
                if otherWantsToSkip && !player.wantsToSkip {
                    Text("(对方已请求)")
                        .font(.system(size: 10))
                }
            }
            .foregroundColor(player.wantsToSkip ? .gold : Color(white: 0.5))
            .padding(.horizontal, 10)
            .padding(.vertical, 5)
            .background(
                RoundedRectangle(cornerRadius: 8)
                    .fill(player.wantsToSkip ? Color.gold.opacity(0.12) : Color.bgCard)
            )
            .overlay(
                RoundedRectangle(cornerRadius: 8)
                    .stroke(player.wantsToSkip ? Color.gold.opacity(0.4) : Color.borderDim, lineWidth: 1)
            )
        }
        .padding(.trailing, 4)
    }

    // MARK: - Round & Timer

    private var roundTimerBar: some View {
        HStack(spacing: 12) {
            Text("第 \(game.currentRound) 轮 / 共 \(game.totalRounds) 轮")
                .font(.system(size: 13, weight: .semibold))
                .foregroundColor(.textDim)

            if game.timeLimit > 0 {
                HStack(spacing: 4) {
                    Image(systemName: "clock")
                        .font(.system(size: 11))
                        .foregroundColor(.textDim)
                    Text("\(game.timeLeft)")
                        .font(.system(size: 18, weight: .heavy).monospacedDigit())
                        .foregroundColor(game.timeLeft <= 10 ? .errorRed : .gold)
                        .opacity(game.timeLeft <= 10 ? (game.timeLeft % 2 == 0 ? 1 : 0.4) : 1)
                        .animation(.easeInOut(duration: 0.3), value: game.timeLeft)
                }
            } else {
                HStack(spacing: 4) {
                    Image(systemName: "clock")
                        .font(.system(size: 11))
                        .foregroundColor(.textDim)
                    Text("∞")
                        .font(.system(size: 18, weight: .heavy))
                        .foregroundColor(.gold)
                }
            }
        }
        .frame(maxWidth: 560)
        .padding(.vertical, 2)
    }

    // MARK: - Expression Display

    private var expressionDisplay: some View {
        Group {
            if player.tokens.isEmpty {
                Text("点击数字和运算符组成算式")
                    .font(.system(size: 14))
                    .foregroundColor(Color(white: 0.17))
            } else {
                buildExpressionText()
                    .lineLimit(1)
                    .minimumScaleFactor(0.6)
            }
        }
        .frame(maxWidth: 560)
        .frame(minHeight: 46)
        .padding(.horizontal, 14)
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(Color.bgInput)
        )
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .stroke(expressionBorderColor, lineWidth: 2)
        )
        .modifier(ShakeEffect(animatableData: CGFloat(player.shakeCount)))
        .animation(.easeInOut(duration: 0.35), value: player.shakeCount)
    }

    private var expressionBorderColor: Color {
        if player.showSuccess { return .successGreen }
        if player.feedback != nil && !player.passed { return .errorRed }
        return .borderDim
    }

    private func buildExpressionText() -> Text {
        player.tokens.reduce(Text("")) { result, token in
            result + tokenText(token)
        }
    }

    private func tokenText(_ token: Token) -> Text {
        switch token {
        case .number:
            return Text(token.display)
                .font(.system(size: 26, weight: .bold, design: .monospaced))
                .foregroundColor(.tokenNum)
        case .op:
            return Text(token.display)
                .font(.system(size: 26, design: .monospaced))
                .foregroundColor(.tokenOp)
        case .paren:
            return Text(token.display)
                .font(.system(size: 26, design: .monospaced))
                .foregroundColor(.tokenParen)
        }
    }

    // MARK: - Feedback

    private var feedbackText: some View {
        Text(player.feedback ?? " ")
            .font(.system(size: 13))
            .foregroundColor(.errorRed)
            .opacity(player.feedback != nil ? 1 : 0)
            .frame(height: 17)
            .animation(.easeOut(duration: 0.2), value: player.feedback)
    }

    // MARK: - Number Buttons

    private var numberButtons: some View {
        HStack(spacing: 10) {
            ForEach(0..<4, id: \.self) { i in
                numberCardButton(index: i)
            }
        }
    }

    private func numberCardButton(index: Int) -> some View {
        let used = player.usedCards[safe: index] ?? false
        let canInput = game.canInputNumber(player: playerIndex)
        let disabled = used || !canInput
        let number = game.numbers[safe: index] ?? 0
        let suit = game.suits[safe: index] ?? .spade

        return Button {
            game.handleNumber(player: playerIndex, cardIndex: index)
        } label: {
            ZStack {
                RoundedRectangle(cornerRadius: 12)
                    .fill(
                        LinearGradient(
                            colors: [
                                Color(red: 28/255, green: 34/255, blue: 64/255),
                                Color(red: 20/255, green: 26/255, blue: 46/255)
                            ],
                            startPoint: .topLeading,
                            endPoint: .bottomTrailing
                        )
                    )
                RoundedRectangle(cornerRadius: 12)
                    .stroke(accent.opacity(0.15), lineWidth: 2)

                VStack(spacing: 0) {
                    Text(suit.symbol)
                        .font(.system(size: 10))
                        .opacity(0.45)
                    Text("\(number)")
                        .font(.system(size: 30, weight: .bold))
                        .foregroundColor(.white)
                }
            }
            .frame(width: 76, height: 96)
            .opacity(used ? 0.18 : (disabled ? 0.35 : 1))
            .scaleEffect(used ? 0.9 : 1)
            .animation(.easeOut(duration: 0.15), value: used)
            .animation(.easeOut(duration: 0.1), value: canInput)
        }
        .disabled(disabled)
    }

    // MARK: - Operator Buttons

    private var operatorButtons: some View {
        let canOp = game.canInputOp(player: playerIndex)
        let canOpen = game.canInputOpenParen(player: playerIndex)
        let canClose = game.canInputCloseParen(player: playerIndex)

        return HStack(spacing: 8) {
            ForEach([("+", "+"), ("-", "−"), ("*", "×"), ("/", "÷"), ("(", "("), (")", ")")], id: \.0) { op, label in
                let enabled: Bool = {
                    switch op {
                    case "(": return canOpen
                    case ")": return canClose
                    default: return canOp
                    }
                }()

                Button {
                    game.handleOp(player: playerIndex, op: op)
                } label: {
                    Text(label)
                        .font(.system(size: 23))
                        .foregroundColor(Color(red: 152/255, green: 152/255, blue: 216/255))
                        .frame(width: 54, height: 48)
                        .background(RoundedRectangle(cornerRadius: 10).fill(Color.bgCard))
                        .overlay(RoundedRectangle(cornerRadius: 10).stroke(Color.borderDim, lineWidth: 2))
                        .opacity(enabled ? 1 : 0.3)
                }
                .disabled(!enabled)
            }
        }
        .animation(.easeOut(duration: 0.1), value: canOp)
        .animation(.easeOut(duration: 0.1), value: canOpen)
        .animation(.easeOut(duration: 0.1), value: canClose)
    }

    // MARK: - Action Buttons

    private var otherWantsToSkip: Bool {
        game.players[1 - playerIndex].wantsToSkip
    }

    private var actionButtons: some View {
        HStack(spacing: 7) {
            actionBtn("提交", color: .successGreen, minW: 88) {
                game.handleSubmit(player: playerIndex)
            }
            actionBtn("清空", color: Color(white: 0.47)) {
                game.handleClear(player: playerIndex)
            }
            actionBtn("⌫", color: Color(white: 0.47)) {
                game.handleBackspace(player: playerIndex)
            }
        }
    }

    private func actionBtn(
        _ title: String,
        color: Color,
        minW: CGFloat = 70,
        action: @escaping () -> Void
    ) -> some View {
        Button(action: action) {
            Text(title)
                .font(.system(size: 14, weight: .semibold))
                .foregroundColor(color)
                .frame(minWidth: minW, minHeight: 48)
                .background(RoundedRectangle(cornerRadius: 10).fill(Color.bgCard))
                .overlay(
                    RoundedRectangle(cornerRadius: 10)
                        .stroke(title == "提交" ? color.opacity(0.35) : Color.borderDim, lineWidth: 2)
                )
        }
    }

    // MARK: - Round Over Overlay

    private var roundOverlay: some View {
        VStack(spacing: 6) {
            if let winner = game.roundWinner {
                if winner == playerIndex {
                    Text("正确！+1 分")
                        .font(.system(size: 30, weight: .heavy))
                        .foregroundColor(.successGreen)
                } else {
                    Text("对手先答出")
                        .font(.system(size: 30, weight: .heavy))
                        .foregroundColor(.errorRed)
                    if let sol = game.solutions.first {
                        Text(sol)
                            .font(.system(size: 15))
                            .foregroundColor(.textDim)
                    }
                }
            } else {
                Text("本轮结束")
                    .font(.system(size: 30, weight: .heavy))
                    .foregroundColor(.gold)
                if let sol = game.solutions.first {
                    Text("答案: \(sol)")
                        .font(.system(size: 15))
                        .foregroundColor(.textDim)
                }
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(Color.bgMain.opacity(0.7))
        .transition(.opacity)
    }

    // MARK: - Background

    private var playerBackground: some View {
        RadialGradient(
            colors: [accent.opacity(0.025), .clear],
            center: .bottom,
            startRadius: 0,
            endRadius: 400
        )
    }
}
